lib/plugins/processor/post.js

```diff
@@ -22,7 +22,7 @@ const preservedKeys = {
 };
 
 module.exports = ctx => {
-  function processPost(file) {
+  async function processPost(file) {
```

Comment:
Avoid using `async` here. The generation performance has been dropped by 30% after the PR.

Reply:
That's strange, how  did you check it? There should not be suck an impact - at least, not at node 12+.

Reply:
https://github.com/hexojs/hexo/issues/3328#issuecomment-666896317

Unit tests have been (mostly) [refactored](https://github.com/hexojs/hexo/pulls?q=is%3Apr+async%2Fawait+is%3Aclosed+) to async/await, refactoring hexo core has been considered before (https://github.com/hexojs/hexo/pull/3952, https://github.com/hexojs/hexo/pull/3970, https://github.com/hexojs/hexo/pull/4019), but we've decided to retain Bluebird promise for a foreseeable time due to its unmatched performance.

Reply:
Perfomance isn't so unmatched in 12-14 versions of node. And promises themselves are just a tiny portion of performance which is often only visible in synthetic benchmarks. Using async await could let you write cleaner code and really raise performance. Also, it seems like soon there will be no developers who know about promises :)